index.js

```diff
@@ -99,7 +100,7 @@ Request.prototype._tryUntilFail = function () {
       response.attempts = this.attempts;
     }
     if (this.retryStrategy(err, response, body) && this.maxAttempts > 0) {
-      this._timeout = setTimeout(this._tryUntilFail.bind(this), this.retryDelay);
+      this._timeout = setTimeout(this._tryUntilFail.bind(this), this.delayStrategy(err, response, body));
```

Comment:
you should do this.delayStrategy.call(this, err, response, body); to make the fallback work